tests/lightgbm/test_lightgbm_autolog.py

```diff
@@ -642,3 +642,38 @@ def test_callback_func_is_pickable():
         functools.partial(_autolog_callback, BatchMetricsLogger(run_id="1234"), eval_results={})
     )
     pickle.dumps(cb)
+
+
+@pytest.mark.large
+@pytest.mark.parametrize("registered_model_name", [None, "model_abc"])
+def test_sklearn_api_autolog_registering_model(registered_model_name):
+    mlflow.lightgbm.autolog(registered_model_name=registered_model_name)
+
+    X, y = datasets.load_iris(return_X_y=True)
+    params = {"n_estimators": 10, "reg_lambda": 1}
+    model = lgb.LGBMClassifier(**params)
+
+    with patch("mlflow.register_model") as mock_register_model, mlflow.start_run():
+        model.fit(X, y)
+
+        if registered_model_name is None:
+            mock_register_model.assert_not_called()
+        else:
+            mock_register_model.assert_called_once_with(
+                ANY, registered_model_name, await_registration_for=ANY
+            )
```

Comment:
why not check the model is registered with the specified name?

Reply:
that might requires more code to write, and `register_model` functionality should be covered by the model registry testing code.

Reply:
> that might require more code to write.

Are you sure? Can we just call `get_registered_model`? Let me know if I'm missing something.

Reply:
> `register_model` functionality should be covered by the model registry testing code.

Yes, but how do we ensure we don't pass incorrect arguments?

Reply:
updated using `MlflowClient().get_registered_model(name)` for test